designer/server/src/main/scala/pl/touk/nussknacker/ui/api/description/scenarioActivity/Dtos.scala

```diff
@@ -0,0 +1,709 @@
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List("compare")
+    }
+
+    case object ScenarioNameChanged extends ScenarioActivityType {
+      override def displayableName: String        = "Scenario name changed"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object CommentAdded extends ScenarioActivityType {
+      override def displayableName: String        = "Comment"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List("delete_comment", "edit_comment")
+    }
+
+    case object AttachmentAdded extends ScenarioActivityType {
+      override def displayableName: String        = "Attachment"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object ChangedProcessingMode extends ScenarioActivityType {
+      override def displayableName: String        = "Processing mode change"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object IncomingMigration extends ScenarioActivityType {
+      override def displayableName: String        = "Incoming migration"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List("compare")
+    }
+
+    case object OutgoingMigration extends ScenarioActivityType {
+      override def displayableName: String        = "Outgoing migration"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object PerformedSingleExecution extends ScenarioActivityType {
+      override def displayableName: String        = "Processing data"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object PerformedScheduledExecution extends ScenarioActivityType {
+      override def displayableName: String        = "Processing data"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object AutomaticUpdate extends ScenarioActivityType {
+      override def displayableName: String        = "Automatic update"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List("compare")
+    }
+
+    override def values: immutable.IndexedSeq[ScenarioActivityType] = findValues
+
+  }
+
+  implicit def scenarioActivityTypeSchema: Schema[ScenarioActivityType] =
```

Comment:
What do you think if codecs would be placed in companion object of the main object? For me it's more readable if a codec is in scope of given object and everything related to that object is in that scope.